pilot/pkg/model/jwks_resolver.go

```diff
@@ -432,7 +428,9 @@ func (r *JwksResolver) refresh() bool {
 		}
 
 		oldPubKey := e.pubKey
-
+		if oldPubKey == "" {
+			oldKeyNull = true
+		}
```

Comment:
In line 441-450, you should update the cache for the jwksURI returned by the resolveJwksURIUsingOpenID() otherwise it will be called every time for an empty jwksURI.

Reply:
one confusion - in the case when jwksuri is not returning the valid public key, do we need to update the jwksuri in the cache with empty public key for next retries ?